WordPress/src/main/java/org/wordpress/android/ui/sitecreation/usecases/CreateSiteUseCase.kt

```diff
@@ -33,9 +35,21 @@ class CreateSiteUseCase @Inject constructor(
         if (continuation != null) {
             throw IllegalStateException("Create site request has already been sent.")
         }
+        /*
+         * To create a site with WordPress.com sub-domain, we need to pass the domain name without the "wordpress.com"
+         * whereas to create a site with other domains, we need to pass the full url. This issue is addressed
+         * in this use case since it's closest to the network layer.
+         *
+         * Ideally API wouldn't work like this or if it does FluxC is the one that handles the issue. However, at the
+         * time of this comment, changing FluxC's Payload might end up affecting the old site creation flow,
+         * so the workaround is applied here instead.
+         */
+        val domain = if (siteData.domain.endsWith(".wordpress.com")) {
```

Comment:
I am not very happy about having to do this in this layer, but I don't think it makes sense to have it in the ViewModel either. At least here, there is some context about why we have to do it and it's clear that it's an API/FluxC thing. We might update FluxC payload later on to move this change there, but since we are targeting the release branch, I think it's best to do it here for now.

For full disclosure, I did try adding a sealed class for the `NewSiteCreationServiceData.domain` where the domain could be either a WordPress.com or .blog subdomain, but that forces view models to know that there is a difference between them and manually create the correct type. Since the domains don't expose which type of domain they are, that forces us to make the `endsWith(".wordpress.com")` to the view model which brings us to square one.

P.S: This is a great example of why proper API is very important, unless it's handled gracefully from the start (in FluxC's network layer) it's effects can leak in a bunch of layers.